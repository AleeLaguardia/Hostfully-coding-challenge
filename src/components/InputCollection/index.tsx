import React from "react";
import {
  Container,
  DestinationInputContainer,
  PeopleInputContainer,
  DateInputContainer,
  ButtonContainer,
  Button,
  DisplayElement,
  CalendarContainer,
} from "./style";
import Input from "../Input";
import CalendarComponent from "../CalendarComponent";
import useClickOutside from "../../utils/hooks/useClickOutside";
import Dropdown from "../Dropdown";

interface Props {
  destination: string;
  startDate: string;
  endDate: string;
  people: string;
  onDestinationChange: React.ChangeEventHandler<HTMLInputElement>;
  confirmChanges: () => void;
  isCalendarOpen: boolean;
  setIsCalendarOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

const InputCollection: React.FC<Props> = ({
  isCalendarOpen,
  setIsCalendarOpen,
  destination,
  startDate,
  endDate,
  people,
  onDestinationChange,
  confirmChanges,
}) => {
  const calendarRef = useClickOutside(() => setIsCalendarOpen(false));

  return (
    <Container>
      <DestinationInputContainer>
        <Input
          error={false}
          placeholder="Destination..."
          onChange={onDestinationChange}
          value={destination}
        />
      </DestinationInputContainer>
      <DateInputContainer
        ref={calendarRef}
        onClick={() => setIsCalendarOpen(true)}
        data-testid="date-input-container"
      >
        <DisplayElement>
          <span>{`${startDate} - ${endDate}`}</span>
        </DisplayElement>
        {isCalendarOpen && (
          <CalendarContainer>
            <CalendarComponent isOpen={isCalendarOpen} setIsOpen={setIsCalendarOpen} />
          </CalendarContainer>
        )}
      </DateInputContainer>
      <PeopleInputContainer>
        <DisplayElement>
          <span>{people.toLowerCase()}</span>
        </DisplayElement>
      </PeopleInputContainer>
    </Container>
  );
};

export default InputCollection;
