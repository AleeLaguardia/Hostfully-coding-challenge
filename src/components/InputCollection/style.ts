import styled from "styled-components";

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-areas: 'destination date people';
  grid-gap: 10px;
  height: 60px;
  width: 900px;
  background-color: ${(props) => props.theme.colors.purple};
  border-radius: 10px;
  padding: 0 10px;
  box-shadow: 0px 10px 5px 0px rgba(0,0,0,0.2);

  @media (max-width: 1000px) {
    width: 95vw;
    margin: 0 20px;
  }
`;

export const DestinationInputContainer = styled.div`
  grid-area: destination;
  height; 100%;
  width: 100%;
`;

export const DateInputContainer = styled.div`
  grid-area: date;
  height; 100%;
  width: 100%;
`;

export const DisplayElement = styled.div<{ backgroundColor?: string }>`
  display: flex;
  align-items: center;
  height: 40px;
  width: 100%;
  padding: 0 10px;
  background-color: ${(props) => props.backgroundColor ? props.backgroundColor : props.theme.colors.white};
  border-radius: 5px;
  margin: 10px 0;
  cursor: pointer;

  span {
    color: ${(props) => props.theme.colors.purple};
    font-family: ${(props) => props.theme.fonts.poppins};
    font-size: ${(props) => props.theme.sizes.xs};
  }
`;

export const PeopleInputContainer = styled.div`
  grid-area: people;
  height; 100%;
  width: 100%;
`;

export const ButtonContainer = styled.div`
  grid-area: button;
  height; 100%;
  width: 100%;
`;

export const Button = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;
  width: 100%;
  background-color: ${(props) => props.theme.colors.selectiveYellow};
  border-radius: 5px;
  border: none;
  cursor: pointer;
  margin: 10px 0;

  &:hover {
    background-color: ${(props) => props.theme.colors.chelseaGem};
  }

  span {
    font-family: ${(props) => props.theme.fonts.poppins};
    font-weight: 500;
    color: ${(props) => props.theme.colors.white};
    font-size: ${(props) => props.theme.sizes.s};
    text-transform: uppercase;
  }
`;

export const CalendarContainer = styled.div`
  position: absolute;
  display: flex;
  height: fit-content;
  width: fit-content;
  box-shadow: 0px 0px 10px 2px rgba(0,0,0,0.3);
  border-radius: 40px;
  margin-top: 10px;
  left: 50px;
`;