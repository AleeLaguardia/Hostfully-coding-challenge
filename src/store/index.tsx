import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slice/userInfoSlice';
import reservationReducer from "./slice/reservationSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    reservation: reservationReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;