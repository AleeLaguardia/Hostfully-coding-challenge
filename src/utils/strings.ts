export const PATH = {
  HOME: '/',
  BOOKING: '/booking',
  RESERVATION: '/reservation',
};

export const HOME = {
  TITLE_FORM: "Book your stay right now!",
  TITLE_PIC: "Book Your Stay",
  ADULTS: "How many adults",
  CHILDREN: "How many children",
}

export const BOOKING = {
  NO_DATA: "No hotels found in that area",
  ALSO_TRY: "But you can also try:"
};