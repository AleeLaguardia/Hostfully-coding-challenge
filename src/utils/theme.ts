import { DefaultTheme } from "styled-components";

import {
  white,
  alabaster,
  emperor,
  galery,
  gray,
  black,
  purple,
  purpleHover,
  darkPurple,
  radicalRed,
  lightPurple,
  silverChalice,
  alto,
  lynch,
  selectiveYellow,
  chelseaGem,
} from './styling/colors';
import {
  poppins,
  playwrite,
} from './styling/fonts';
import {
  xs,
  xxs,
  xxxs,
  s,
  m,
  l,
  xl,
  xxxl,
} from './styling/sizes';

export const theme: DefaultTheme = {
  colors: {
    white,
    alabaster,
    emperor,
    galery,
    gray,
    black,
    purple,
    purpleHover,
    darkPurple,
    radicalRed,
    lightPurple,
    silverChalice,
    alto,
    lynch,
    selectiveYellow,
    chelseaGem,
  },
  fonts: {
    poppins,
    playwrite,
  },
  sizes: {
    xs,
    xxs,
    xxxs,
    s,
    m,
    l,
    xl,
    xxxl,
  },
}