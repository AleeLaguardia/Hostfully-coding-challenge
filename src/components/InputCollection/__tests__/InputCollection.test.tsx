import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import InputCollection from "..";
import { ThemeProvider } from "styled-components";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { theme } from "../../../utils/theme";
import store from "../../../store";

const mockProps = {
  destination: "Paris",
  startDate: "01/01/2022",
  endDate: "01/10/2022",
  people: "2 adults",
  onDestinationChange: jest.fn(),
  confirmChanges: jest.fn(),
  isCalendarOpen: false,
  setIsCalendarOpen: jest.fn(),
};

const renderWithProviders = (ui: React.ReactElement) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>
        <ThemeProvider theme={theme}>{ui}</ThemeProvider>
      </BrowserRouter>
    </Provider>
  );
};

describe("InputCollection Component", () => {
  it("renders with default styling", () => {
    const { asFragment } = renderWithProviders(<InputCollection {...mockProps} />)

    expect(asFragment).toMatchSnapshot();
  })

  it("renders correctly with given props", () => {
    renderWithProviders(<InputCollection {...mockProps} />);
    
    
    expect(screen.getByPlaceholderText("Destination...")).toHaveValue("Paris");
    
    
    expect(screen.getByText("01/01/2022 - 01/10/2022")).toBeInTheDocument();
    
    
    expect(screen.getByText("2 adults")).toBeInTheDocument();
  });

  it("calls onDestinationChange when destination input changes", () => {
    renderWithProviders(<InputCollection {...mockProps} />);
    
    const destinationInput = screen.getByPlaceholderText("Destination...");
    fireEvent.change(destinationInput, { target: { value: "New York" } });
    
    expect(mockProps.onDestinationChange).toHaveBeenCalled();
  });

  it("toggles calendar visibility when date input container is clicked", () => {
    renderWithProviders(<InputCollection {...mockProps} />);
    
    const dateInputContainer = screen.getByTestId("date-input-container");
    fireEvent.click(dateInputContainer);
    
    expect(mockProps.setIsCalendarOpen).toHaveBeenCalledWith(true);
  });

  it("displays the calendar when isCalendarOpen is true", () => {
    renderWithProviders(<InputCollection {...mockProps} isCalendarOpen={true} />);
    
    expect(screen.getByTestId("calendar-component")).toBeInTheDocument();
  });

  it("closes the calendar when clicking outside of it", () => {
    renderWithProviders(<InputCollection {...mockProps} isCalendarOpen={true} />);
    
    
    const dateInputContainer = screen.getByTestId("date-input-container");
    fireEvent.mouseDown(dateInputContainer);
    expect(mockProps.setIsCalendarOpen).not.toHaveBeenCalledWith(false);

    
    fireEvent.mouseDown(document);
    expect(mockProps.setIsCalendarOpen).toHaveBeenCalledWith(false);
  });
});
