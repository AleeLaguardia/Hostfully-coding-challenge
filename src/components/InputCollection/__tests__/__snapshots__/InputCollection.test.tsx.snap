// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCollection Component renders with default styling 1`] = `[Function]`;
